scripts/coloring.js

```diff
@@ -14,7 +14,21 @@ const coloring = {
         const preNodes = document.querySelectorAll("body > div > pre");
 
         for (const pre of preNodes) {
-            for (let i = 0; i < pre.childNodes.length; i++) {
+            let i, j = -1;
+
+            for (i = 0; i < pre.childNodes.length && j < 0; i++) {
+                if (pre.childNodes[i].nodeName === "#text") {
+                    /*
+                     * man git-format-patch: The log message and the patch are
+                     * separated by a line with a three-dash line.
+                     */
+                    j = pre.childNodes[i].textContent.search(/^---$/m);
```

Comment:
I'm slightly concerned that this might be a bit large and catch other things than the `git format-patch` separator. I'm thinking that Markdown files can use `^---$` as separators, for example. On the other hand, if we end up colouring a raw Markdown file with diff syntax, it will be messed up anyway. So yeah, maybe we can do that and see if anyone complains that it break their patches.